tests/unit/ts_utils_spec.lua

```diff
@@ -106,3 +106,35 @@ describe("update_selection", function()
     )
   end)
 end)
+
+describe("swap_nodes", function()
+  local function swap_and_return_cursor(case)
+    vim.api.nvim_buf_set_lines(0, 0, -1, false, case.lines)
+    local a = vim.treesitter.get_node_at_position(0, case.a[1], case.a[2], {})
+    local b = vim.treesitter.get_node_at_position(0, case.b[1], case.b[2], {})
+    tsutils.swap_nodes(a, b, 0, true)
+    return vim.api.nvim_win_get_cursor(vim.api.nvim_get_current_win())
+  end
+
+  it("moves cursor correctly with adjacent nodes", function()
+    assert.equal(
+      swap_and_return_cursor {
+        lines = { "print(1)" },
+        a = { 0, 0 },
+        b = { 0, 5 },
+      },
+      { 1, 3 }
```

Comment:
ok, cursor position is interesting. Wouldn't it be interesting to assert on the swapped text also?

Reply:
The bug specifically is only involved with the location of the cursor after the movement, but yes it would be smart to add the text content to the assert as well